src/layout/rectangles.js

```diff
@@ -58,7 +62,7 @@ export default function(layoutStrategy) {
             // filter the data to include only those remaining after collision removal
             var filteredData = filteredLayout.map(function(d) { return data[d.dataIndex]; });
 
-            var g = dataJoin(this, filteredData);
+            g = dataJoin(this, filteredData);
```

Comment:
That's a good idea, by hiding filtered labels I can avoid a second join - this also means that components that are rendered via this layout can be a little simpler, i.e. they don't need to respect enter / update.